src/Autofac.Extensions.DependencyInjection/AutofacRegistration.cs

```diff
@@ -47,14 +47,21 @@ public static class AutofacRegistration
         /// <param name="descriptors">
         /// The set of service descriptors to register in the container.
         /// </param>
+        /// <param name="lifetimeScopeTagForSingletons">
+        /// If provided and not <c>null</c>then all registrations with lifetime <see cref="ServiceLifetime.Singleton" /> are registered
+        /// using <see cref="IRegistrationBuilder{TLimit,TActivatorData,TRegistrationStyle}.InstancePerMatchingLifetimeScope" />
+        /// with provided <paramref name="lifetimeScopeTagForSingletons"/>
+        /// instead of using <see cref="IRegistrationBuilder{TLimit,TActivatorData,TRegistrationStyle}.SingleInstance"/>.
+        /// </param>
         public static void Populate(
-                this ContainerBuilder builder,
-                IEnumerable<ServiceDescriptor> descriptors)
+            this ContainerBuilder builder,
+            IEnumerable<ServiceDescriptor> descriptors,
+            object lifetimeScopeTagForSingletons = null)
```

Comment:
Instead of adding an optional parameter, create a new method overload. Turns out changing the signature, even with an optional parameter, is a breaking interface change.